src/main/java/gregtech/common/covers/filter/IFilter.java

```diff
@@ -0,0 +1,66 @@
+package gregtech.common.covers.filter;
+
+import gregtech.api.util.IDirtyNotifiable;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+
+import com.cleanroommc.modularui.api.widget.IWidget;
+import com.cleanroommc.modularui.screen.ModularPanel;
+import com.cleanroommc.modularui.value.sync.GuiSyncManager;
+import com.cleanroommc.modularui.widget.Widget;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Consumer;
+
+public interface IFilter {
+
+    @Deprecated
+    void initUI(Consumer<gregtech.api.gui.Widget> widgetGroup);
+
+    /** Uses Cleanroom MUI */
+    @NotNull
+    ModularPanel createPopupPanel(GuiSyncManager syncManager);
+
+    /** Uses Cleanroom MUI */
+    @NotNull
+    ModularPanel createPanel(GuiSyncManager syncManager);
+
+    /** Uses Cleanroom MUI - Creates the widgets standalone so that they can be put into their own panel */
+
+    @NotNull
+    Widget<?> createWidgets(GuiSyncManager syncManager);
+
+    ItemStack getContainerStack();
+
+    void setDirtyNotifiable(@Nullable IDirtyNotifiable dirtyNotifiable);
+
+    void markDirty();
+
+    int getMaxTransferSize();
+
+    void setMaxTransferSize(int maxTransferSize);
+
+    boolean showGlobalTransferLimitSlider();
+
+    default int getTransferLimit(int slot, int transferSize) {
+        return transferSize;
+    }
+
+    boolean isBlacklistFilter();
+
+    void setBlacklistFilter(boolean blacklistFilter);
+
+    IWidget createBlacklistUI();
```

Comment:
Does this really need to be in the interface? Or can it simply be in `BaseFilter`, since things all extend from that anyway.

Reply:
not really, moved to BaseFilter